pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -57,7 +57,11 @@ func defaultPredicates() sets.String {
 }
 
 // ApplyFeatureGates applies algorithm by feature gates.
-func ApplyFeatureGates() {
+// The returned function is used to restore the state of registered predicates/priorities
+// when this function is called, and should be called in tests where may modify the value
+// of feature gate temporarily.
+func ApplyFeatureGates() func() {
```

Comment:
It might be better to give a name for `func()`, e.g. `RestoreXXX`?

Reply:
+1

Reply:
Add `restore` as a named return function.